### PR TITLE
Adding test scripts (YAMLs) for Laundry Dryer Mode

### PR DESCRIPTION
**Pull request by prakashece**
The test scripts are based on LaundryDryerControls.adoc. Please find the link below to access it.

https://github.com/CHIP-Specifications/chip-test-plans/pull/3686

This is the link of Older approved PR for the same https://github.com/project-chip/connectedhomeip/pull/30716. I closed it and created this branch to resolve the CLA issues.

### CODE REVIEW

File: src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+PICS:
+    - DRYERCTRL.S
+
+config:
+    nodeId: 0x12344321
+    cluster: "Laundry Dryer Controls"
+    endpoint: 1
+
+    unsupportedSelectedDrynessLevel:
```
[the remark this answers is not in the document]
Comment (cecille) [reply src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:25]: oh, so 4 is an out of range enum. Is the server meant to reply to an out of range value the same way as an unsupported value?

File: src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    - label:
+          "Step 6: TH writes an unsupported SelectedDrynessLevel attribute that
+          is not in list of SupportedDrynessLevels to DUT"
+      PICS: DRYERCTRL.S.A0001
+      command: "writeAttribute"
+      attribute: "SelectedDrynessLevel"
+      arguments:
+          value: unsupportedSelectedDrynessLevel
+      response:
+          error: INVALID_IN_STATE
```
Comment (cecille) [inline src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:102]: Is this the correct error? Per the spec - 

If an attempt is made to write this attribute with a value other than null or a value contained in SupportedDrynessLevels, a CONSTRAINT_ERROR response SHALL be sent as the response

File: src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (prakashece) [reply src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:25]: Yes, correct

File: src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml (hunk 2 above)
Comment (prakashece) [reply src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:102]: I see what you are saying, but I followed the Dryer control test plan https://github.com/CHIP-Specifications/chip-test-plans/pull/3686 after it was merged.

Comment (bzbarsky-apple) [reply src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:102]: It looks to me like both the test plan and the SDK implementation are wrong here, in the sense of not matching the spec....

Comment (prakashece) [reply src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:102]: A Test plan PR has been created to address the above issue https://github.com/CHIP-Specifications/chip-test-plans/pull/3808

Comment (prakashece) [reply src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:126]: @cecille @bzbarsky-apple  can you pls review the PR? I have made the necessary update based on the new TP https://github.com/CHIP-Specifications/chip-test-plans/pull/3808

**Review by bzbarsky-apple — APPROVED** [review]
Approving, but:

1.  I did not carefully compare this YAML to the test plan.  Just making sure I am not blocking this.
2. The test is failing as expected, because the implementation is broken.  The implementation needs to be fixed before this can land, right?

**Comment by prakashece** [comment]
> Approving, but:
> 
> 1. I did not carefully compare this YAML to the test plan.  Just making sure I am not blocking this.
> 2. The test is failing as expected, because the implementation is broken.  The implementation needs to be fixed before this can land, right?
 Yes a PR has been created on the implementation side to pass the Test scripts.

**Review by cecille — APPROVED** [review]

File: src/app/tests/suites/certification/ci-pics-values [hunk 3]
```diff
@@ -750,6 +750,22 @@ PIXIT.LWM.MODE_CHANGE_OK=1
...
+
+# LAUNDRY DRYER MODE CLUSTER
+DRYERCTRL.S=1
+DRYERCTRL.S.A0000=1
+DRYERCTRL.S.A0001=1
+
+#Feature
+DRYERCTRL.S.F00=1
+
+#PIXIT
```
Comment (cecille) [inline src/app/tests/suites/certification/ci-pics-values:762]: Could you please remove the PIXITs from the file?

**Pull request merged**